File: 0969_pancake_sorting/pancake.java
import java.util.*;

class Solution {
    // Time: O(n**2)
    // Space: O(n)
    public List<Integer> pancakeSort(int[] A) {
        List<Integer> result = new ArrayList<>();
        int n = A.length, largest = n;
        for (int i = 0; i < n; i++) {
            int index = find(A, largest);
            if (index != 0) {
                flip(A, index);
                result.add(index + 1);
            }
            flip(A, largest - 1);
            result.add(largest--);
        }
        return result;
    }

    private int find(int[] A, int target) {
        for (int i = 0; i < A.length; i++) {
            if (A[i] == target) {
                return i;
            }
        }
        return -1;
    }

    private void flip(int[] A, int index) {
        int i = 0, j = index;
        while (i < j) {
            int temp = A[i];
            A[i++] = A[j];
            A[j--] = temp;
        }
    }
}

File: 0969_pancake_sorting/pancake.py
from typing import List


def pancake_sort(A: List[int]) -> List[int]:
    """
    Time: O(n**2)
    Space: O(n)
    """
    res = []
    for x in range(len(A), 1, -1):
        i = A.index(x)
        res.extend([i + 1, x])
        A = A[:i:-1] + A[:i]
    return res
